nifi-nar-bundles/nifi-ccda-bundle/nifi-ccda-processors/src/main/java/org/apache/nifi/processors/ccda/ExtractCCDAAttributes.java

```diff
@@ -0,0 +1,329 @@
+            .name("success")
+            .description("A FlowFile is routed to this relationship if it is properly parsed as CDA and its content stored in FlowFile")
+            .build();
+
+    /**
+     * REL_FAILURE - Value to be returned in case the processor fails
+     */
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if it cannot be parse CDA and store content to FlowFile.")
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+
+        relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+
+        properties = new ArrayList<>();
+        properties.add(SKIP_VALIDATION);
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) throws IOException {
+        getLogger().debug("Loading packages");
+        final StopWatch stopWatch = new StopWatch(true);
+
+        // Load required MDHT packages
+        System.setProperty( "org.eclipse.emf.ecore.EPackage.Registry.INSTANCE",
+                "org.eclipse.emf.ecore.impl.EPackageRegistryImpl" );
+        CDAPackage.eINSTANCE.eClass();
+        HITSPPackage.eINSTANCE.eClass();
+        CCDPackage.eINSTANCE.eClass();
+        ConsolPackage.eINSTANCE.eClass();
+        IHEPackage.eINSTANCE.eClass();
+        stopWatch.stop();
+        getLogger().debug("Loaded packages in {}", new Object[] {stopWatch.getDuration(TimeUnit.MILLISECONDS)});
+
+        // Initialize JEXL
+        jexl = new JexlBuilder().cache(1024).debug(false).silent(true).strict(false).create();
+        jexlCtx = new MapContext();
+
+        getLogger().debug("Loading mappings");
+        loadMappings(); // Load CDA mappings for parser
+
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        Map<String, String> attributes = new TreeMap<String, String>(); // stores CDA attributes
+        getLogger().info("Processing CCDA");
+
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        if(processMap.isEmpty()) {
+            getLogger().error("Process Mapping is not loaded");
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final Boolean skipValidation = context.getProperty(SKIP_VALIDATION).asBoolean();
+
+        final StopWatch stopWatch = new StopWatch(true);
+
+        ClinicalDocument cd = loadDocument(session.read(flowFile), skipValidation); // Load and optionally validate CDA document

```

Comment:
Need to handle the exceptions coming out of here and sending the FlowFile to REL_FAILURE; otherwise they're getting stuck in the queue forever.